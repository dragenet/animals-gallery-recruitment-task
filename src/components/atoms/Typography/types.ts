import { Colors, FontSize } from '../../../theme';

export interface TypographyProps {
  color?: Colors;
  fontSize?: FontSize;
  bold?: boolean;
}
