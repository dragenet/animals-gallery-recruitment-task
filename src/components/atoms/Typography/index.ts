export * from './types';
export * from './Typography';
export * from './utils';
