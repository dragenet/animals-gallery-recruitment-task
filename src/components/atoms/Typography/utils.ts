import { css, ThemedStyledProps } from 'styled-components';

import { Colors, FontSize } from '../../../theme';
import { TypographyProps } from './types';

export const fontSize = (fontSize: FontSize) => css`
  font-size: ${({ theme }) => theme.fontSizes[fontSize]};
`;

export const color = (color: Colors = 'black') => css`
  color: ${(props) => props.theme.colors[color]};
`;

export const bold = css`
  font-weight: bold;
`;

export const typographyMixin =
  (options?: Partial<TypographyProps>) =>
  <T extends TypographyProps, V>(props: ThemedStyledProps<T, V>) => {
    const currentProps = { ...options, ...props };

    return css`
      ${currentProps.fontSize && fontSize(currentProps.fontSize)}
      ${currentProps.color && color(currentProps.color)}
  ${currentProps.bold && bold}
    `;
  };
