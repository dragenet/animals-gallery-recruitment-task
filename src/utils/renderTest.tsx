import { render } from '@testing-library/react';

import { RootProvider } from '~/components/atoms/RootProvider';

export const renderTest = (children: JSX.Element) =>
  render(<RootProvider>{children}</RootProvider>);
