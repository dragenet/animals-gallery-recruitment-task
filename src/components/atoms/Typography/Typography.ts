import styled from 'styled-components';

import { TypographyProps } from './types';
import { typographyMixin } from './utils';

export const H1 = styled.h1<TypographyProps>`
  ${typographyMixin({ fontSize: 'large' })}
`;

export const Subtitle = styled.p<TypographyProps>`
  ${typographyMixin({ fontSize: 'small', bold: true })}
`;

export const Body = styled.p<TypographyProps>`
  ${typographyMixin()}
`;
