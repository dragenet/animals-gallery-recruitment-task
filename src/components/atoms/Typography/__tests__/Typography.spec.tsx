import { describe, expect, it } from 'vitest';

import { renderTest } from '~/utils/renderTest';

import { Body, H1, Subtitle } from '../Typography';

describe('Typography', () => {
  it('H1 renders correctly', () => {
    const { container } = renderTest(<H1>Heading 1</H1>);

    expect(container).toMatchSnapshot();
  });

  it('Subtitle renders correctly', () => {
    const { container } = renderTest(<Subtitle>Subtitle</Subtitle>);

    expect(container).toMatchSnapshot();
  });

  it('Body renders correctly', () => {
    const { container } = renderTest(<Body>Body</Body>);

    expect(container).toMatchSnapshot();
  });
});
