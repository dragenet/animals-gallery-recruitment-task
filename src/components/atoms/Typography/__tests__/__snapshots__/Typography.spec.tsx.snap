// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography > Body renders correctly 1`] = `
<div>
  <p
    class="sc-dmqHEX kWFPzJ"
  >
    Body
  </p>
</div>
`;

exports[`Typography > H1 renders correctly 1`] = `
<div>
  <h1
    class="sc-beqWaB fFhbVn"
  >
    Heading 1
  </h1>
</div>
`;

exports[`Typography > Subtitle renders correctly 1`] = `
<div>
  <p
    class="sc-gueYoa gEQEOZ"
  >
    Subtitle
  </p>
</div>
`;
